lib/src/facts/windows/operating_system_resolver.cc

```diff
@@ -101,7 +104,25 @@ namespace facter { namespace facts { namespace windows {
         auto version = result.release.substr(0, lastDot);
         bool consumerrel = (wmi::get(vals, wmi::producttype) == "1");
         if (version == "10.0") {
-            result.release = consumerrel ? "10" : "2016";
+            // Calculate the build number to distinguish between
+            // Windows Server 2016 and 2019. Note that the kernel
+            // version is written as <major>.<minor>.<build_number>
+            auto kernel_version_fact = facts.get<string_value>(fact::kernel_version);
+            if (! kernel_version_fact) {
+                return result;
```

Comment:
Should a warning/debug message be outputted here? We don't do this in [earlier steps](https://github.com/puppetlabs/facter/blob/master/lib/src/facts/windows/operating_system_resolver.cc#L90-L98) which is why I didn't do it here.

Reply:
I feel a Debug message is required at a minimum.

Reply:
Good point. I'll go ahead and add one.